### PR TITLE
Change the URL to the PHP Date Function

### PR DESCRIPTION
**Pull request by imanickam**
Current:
http://www.php.net/manual/en/function.date.php - this URL does not work; it requires signing into the php.net site

Proposed:
http://php.net/date - this URL works and is consistent with other language strings MOD_ARTICLES_CATEGORY_FIELD_DATEFIELDFORMAT_DESC and MOD_ARTICLES_CATEGORY_FIELD_MONTHYEARFORMAT_DESC

Pull Request for Issue # .

### Summary of Changes

Current language string:
JGLOBAL_DATE_FORMAT_DESC="Optional format string for showing the date. For example, D M Y for day month year or you can use d-m-y for a short version eg. 28-12-16. See http://www.php.net/manual/en/function.date.php. If left blank, it uses DATE_FORMAT_LC1 from your language file."

Revised language string:
JGLOBAL_DATE_FORMAT_DESC="Optional format string for showing the date. For example, D M Y for day month year or you can use d-m-y for a short version eg. 28-12-16. See http://php.net/date. If left blank, it uses DATE_FORMAT_LC1 from your language file."

### Testing Instructions

(a) Go to the screen "Articles: Options" (Content => Articles > Options)
(b) Click on the tab List Layouts
(c) For the option "Show Date", choose the value Created or Modified or Published)
(d) Move the cursor over the label "Date Format" - Review the pop-up help text
(e) Type the URL shown in the pop-up help text in a separate window/tab of a browser and see what happens

### Expected result

(a) Pop-up Help Text

after_language_string_change

(b) URL to the PHP Date Function taken from the pop-up Help Text

php_function-date_url_without_www

### Actual result

(a) Pop-up Help Text

before_language_string_change

(b) URL to the PHP Date Function taken from the pop-up Help Text

php_function-date_url_with_www

### CODE REVIEW

**Comment by PhilETaylor** [comment]
http://php.net/manual/en/function.date.php works fine for me here without authentication.

```
HTTP/1.1 200 OK
Connection: keep-alive
Content-Encoding: gzip
Content-Type: text/html; charset=utf-8
Content-language: en
Date: Thu, 15 Nov 2018 06:20:59 GMT
Last-Modified: Thu, 15 Nov 2018 05:04:46 GMT
Link: <http://php.net/date>; rel=shorturl
Server: nginx/1.10.3
Set-Cookie: LAST_LANG=en; expires=Fri, 15-Nov-2019 06:20:59 GMT; Max-Age=31536000; path=/; domain=.php.net
Set-Cookie: COUNTRY=NA%2C87.244.71.75; expires=Thu, 22-Nov-2018 06:20:59 GMT; Max-Age=604800; path=/; domain=.php.net
Transfer-Encoding: chunked
Vary: Accept-Encoding
X-Frame-Options: SAMEORIGIN
```

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 6cbe0bce95a147e02d8fe19d799428d84cae1e54<br /><br />If this may help some users, let's change it.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23082">issues.joomla.org/tracker/joomla-cms/23082</a>.</sub>

**Comment by PhilETaylor** [comment]
yes I agree, I think the redirect also handles the language param in the url right as well - another reason to change it.

**Comment by PhilETaylor** [comment]
Just being picky but can we use https://secure.php.net/date instead of http://php.net/date ?

**Comment by PhilETaylor** [comment]
also might be good to remove all these https://github.com/joomla/joomla-cms/search?q=php.net%2Fmanual%2Fen&unscoped_q=php.net%2Fmanual%2Fen as well?

**Comment by PhilETaylor** [comment]
I have tested this item :white_check_mark: successfully on 6cbe0bce95a147e02d8fe19d799428d84cae1e54<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23082">issues.joomla.org/tracker/joomla-cms/23082</a>.</sub>

**Comment by infograf768** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23082">issues.joomla.org/tracker/joomla-cms/23082</a>.</sub>

**Comment by brianteeman** [comment]
This is clearly an upstream problem with the mirror that you are being directed to. Have you reported it upstream so it can be fixed for everyone that uses php

File: administrator/language/en-GB/en-GB.ini [hunk 1]
```diff
@@ -342,7 +342,7 @@ JGLOBAL_CREATED_DATE="Created Date"
 JGLOBAL_CUSTOM_CATEGORY="New Categories"
 JGLOBAL_CUSTOM_FIELDS_ENABLE_DESC="Enable the creation of custom fields."
 JGLOBAL_CUSTOM_FIELDS_ENABLE_LABEL="Enable Custom Fields"
-JGLOBAL_DATE_FORMAT_DESC="Optional format string for showing the date. For example, D M Y for day month year or you can use d-m-y for a short version eg. 28-12-16. See http://www.php.net/manual/en/function.date.php. If left blank, it uses DATE_FORMAT_LC1 from your language file."
+JGLOBAL_DATE_FORMAT_DESC="Optional format string for showing the date. For example, D M Y for day month year or you can use d-m-y for a short version eg. 28-12-16. See http://php.net/date. If left blank, it uses DATE_FORMAT_LC1 from your language file."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.ini]:
```suggestion
JGLOBAL_DATE_FORMAT_DESC="Optional format string for showing the date. For example, D M Y for day month year or you can use d-m-y for a short version eg. 28-12-16. See https://php.net/date. If left blank, it uses DATE_FORMAT_LC1 from your language file."
```

**Comment by imanickam** [comment]
> 
> 
> This is clearly an upstream problem with the mirror that you are being directed to. Have you reported it upstream so it can be fixed for everyone that uses php

Yes, the problem has been reported - Bug #77161 -	PHP.net Website Problem (https://bugs.php.net/bug.php?id=77161).

**Comment by PhilETaylor** [comment]
Might be worth telling the PHP guys where you are in the world, and maybe your IP because then they might be able to "guess" which mirror you get redirected to.

**Comment by imanickam** [comment]
> 
> 
> Might be worth telling the PHP guys where you are in the world, and maybe your IP because then they might be able to "guess" which mirror you get redirected to.

Good Idea.  I thought that the IP address would be stored along with the bug report when you create one, just like we can find out from which IP address the post has been created in the Joomla! Forum.

**Comment by imanickam** [comment]
Standardized the URL that references the _date_ function in PHP from http://php.net/date to https://php.net/date.

This has been changed in the file en-GB.mod_articles_category.ini (that is located in the directory \language\en-GB) as well.

The URLs are changed in the language strings MOD_ARTICLES_CATEGORY_FIELD_DATEFIELDFORMAT_DESC and MOD_ARTICLES_CATEGORY_FIELD_MONTHYEARFORMAT_DESC

**Review by brianteeman — APPROVED** [review]

**Pull request merged**